### PR TITLE
upgrade to v6 of CAF module

### PR DESCRIPTION
**Pull request by jaredfholgate**
## Overview/Summary

Update to target v6 of the CAF module

## This PR fixes/adds/changes/removes

N/A

### Breaking Changes

See CAF module release notes

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/alz-terraform-accelerator/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/alz-terraform-accelerator/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/alz-terraform-accelerator/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

**Review by matt-FFFFFF — APPROVED** [review]

**Review by luke-taylor — DISMISSED** [review]

File: templates/hubnetworking/terraform.tf [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+    }
+    azapi = {
+      source  = "Azure/azapi"
+      version = "~> 1.13"
+    }
   }
   # backend "azurerm" {}
 }
 
+provider "azapi" {
```
Comment (jaredfholgate) [inline templates/hubnetworking/terraform.tf:16]: @matt-FFFFFF Should this provider block have `subscription_id = var.subscription_id_management`? I'm going to test this with a customer that is getting auth issues, but wondering if you already hit this issue?

Comment (matt-FFFFFF) [reply templates/hubnetworking/terraform.tf:16]: Hey - it needs rights to deploy MG diag settings plus the AMA stuff

Comment (matt-FFFFFF) [reply templates/hubnetworking/terraform.tf:16]: Therefore the higher level assignment at MG scope should be sufficient for this, depending on the order of operations in moving the sub into the MG

Comment (jaredfholgate) [reply templates/hubnetworking/terraform.tf:16]: I don't think the change made any difference to the customer anyway. I suspect something else is going on in their environment. I have tested with this change though and think it is valid to leave it like this for the accelerator.

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**